policy/policy_handler/handler.go

```diff
@@ -0,0 +1,223 @@
+
+	Handler interface {
+		Start()
+	}
+
+	subscriptionProvider func(ctx context.Context, req skill.RequestContext) ([][]edn.RawMessage, skill.Configuration, error)
+	dataSourceProvider   func(ctx context.Context, req skill.RequestContext) ([]data.DataSource, error)
+	transactionFilter    func(ctx context.Context, req skill.RequestContext) bool
+
+	EventHandler struct {
+		// parameters
+		evalSelector      EvaluatorSelector
+		subscriptionNames []string
+
+		// hooks used by opts
+		subscriptionDataProviders []subscriptionProvider
+		dataSourceProviders       []dataSourceProvider
+		transactFilters           []transactionFilter
+	}
+
+	Opt func(handler *EventHandler)
+)
+
+func NewPolicyEventHandler(subscriptionNames []string, evalSelector EvaluatorSelector, opts ...Opt) EventHandler {
+	p := EventHandler{
+		subscriptionNames: subscriptionNames,
+		evalSelector:      evalSelector,
+	}
+
+	for _, o := range opts {
+		o(&p)
+	}
+
+	return p
+}
+
+func (h EventHandler) Start() {
+	handlers := skill.Handlers{}
+	for _, n := range h.subscriptionNames {
+		handlers[n] = h.handle
+	}
+
+	skill.Start(handlers)
+}
+
+// EvaluateGoals runs the goal evaluation and returns the results after transacting them.
+func (h EventHandler) handle(ctx context.Context, req skill.RequestContext) skill.Status {
+	var (
+		subscriptionResult [][]edn.RawMessage
+		configuration      skill.Configuration
+		err                error
+	)
+	for _, provider := range h.subscriptionDataProviders {
+		subscriptionResult, configuration, err = provider(ctx, req)
+		if err != nil {
+			return skill.NewFailedStatus(fmt.Sprintf("failed to retrieve subscription result [%s]", err.Error()))
+		}
+		if subscriptionResult != nil {
+			break
+		}
+	}
+
+	if subscriptionResult == nil {
+		return skill.NewFailedStatus(fmt.Sprintf("subscription result was not found"))
+	}
+
+	sources := []data.DataSource{}
+	for _, provider := range h.dataSourceProviders {
+		ds, err := provider(ctx, req)
+		if err != nil {
+			return skill.NewFailedStatus(fmt.Sprintf("failed to create data source [%s]", err.Error()))
+		}
+		sources = append(sources, ds...)
+	}
+
+	dataSource := data.NewChainDataSource(sources...)
+
+	return h.evaluate(ctx, req, dataSource, subscriptionResult, configuration)
+}
+
+func (h EventHandler) evaluate(ctx context.Context, req skill.RequestContext, dataSource data.DataSource, subscriptionResult [][]edn.RawMessage, configuration skill.Configuration) skill.Status {
+	goalName := req.Event.Skill.Name
+
+	cfg := configuration.Name
+	params := configuration.Parameters
+
+	paramValues := map[string]interface{}{}
+	for _, p := range params {
+		paramValues[p.Name] = p.Value
+	}
+
+	if _, ok := paramValues["definitionName"]; !ok {
+		return skill.NewFailedStatus("Missing definition name in policy skill configuration")
+	}
+
+	goal := goals.Goal{
+		Definition:    paramValues["definitionName"].(string),
```

Comment:
I think this is out of date - goal definition names == skill names now.

Reply:
This is still being used in `goal-evaluation-skill`, do you know if this is true now / if there's any outstanding work that needs to be done before this is true? I see the subscriptions are still using `definitionName` to source the storage tuples, does that need to change?

I'll update the handler to use `req.Event.Skill.Name` if you think we're good to start using that.

Reply:
Yes, IIRC, `definitionName` should be == to the skill name for all current policy skills

Reply:
ok, done fe861425594ee31c6accada88a5d537298afdd48